src/nvim/api/private/helpers.h

```diff
@@ -94,7 +94,7 @@
 
 #define cbuf_as_string(d, s) ((String) { .data = d, .size = s })
 
-#define STATIC_CSTR_AS_STRING(s) ((String) { .data = s, .size = sizeof(s) - 1 })
+#define STATIC_CSTR_AS_STRING(s) ((String) { .data = s, .size = sizeof("" s) - 1 })
```

Comment:
👀

Reply:
Oh, that just exists to prevent non-static strings from being passed to `STATIC_CSTR_AS_STRING`

Reply:
I guess this is applicable to `S_LEN` and `STATIC_CSTR_TO_STRING` too.

Reply:
It should be, yes

Reply:
@zeertzjq seems like `S_LEN` is used for a lot of `char[]` variables in which case this safeguard shows an error even though `sizeof` does work as intended for `char[]` variables.